cmd/devstream/init.go

```diff
@@ -20,35 +24,119 @@ var initCMD = &cobra.Command{
 	Run:   initCMDFunc,
 }
 
+var (
+	downloadOnly      bool     // download plugins only, from command line flags
+	downloadAll       bool     // download all plugins
+	pluginsToDownload []string // download specific plugins
+	initOS            string   // download plugins for specific os
+	initArch          string   // download plugins for specific arch
+)
+
 func initCMDFunc(_ *cobra.Command, _ []string) {
+	if version.Dev {
+		log.Fatalf("Dev version plugins can't be downloaded from the remote plugin repo; please run `make build-plugin.PLUGIN_NAME` to build them locally.")
+	}
+
+	var (
+		realPluginDir string
+		tools         []configmanager.Tool
+		err           error
+	)
+
+	switch downloadOnly {
```

Comment:
```go
	switch downloadOnly {
 	// download plugins according to the config file
 	case false:
 		tools, realPluginDir, err = GetPluginsAndPluginDirFromConfig()
 	// download plugins from flags
 	case true:
 		tools, realPluginDir, err = GetPluginsAndPluginDirFromFlags()
 	}
```

vs

```go
if downloadOnly {
    tools, realPluginDir, err = GetPluginsAndPluginDirFromFlags()
} else {
    tools, realPluginDir, err = GetPluginsAndPluginDirFromConfig()
}
```

I think the "if" version is more clear here.

Reply:
That sounds great.